platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/events/UnsignedEvent.java

```diff
@@ -161,53 +169,159 @@ public BaseEventHashedData(
-        final TransactionConfig transactionConfig = ConfigurationHolder.getConfigData(TransactionConfig.class);
-        deserialize(in, version, transactionConfig.maxTransactionCountPerEvent());
+    private static void serializeApplicationPayload(@NonNull final SerializableDataOutputStream out, @NonNull final EventPayload payload)
+            throws IOException {
+        out.writeLong(APPLICATION_TRANSACTION_CLASS_ID);
+        out.writeInt(APPLICATION_TRANSACTION_VERSION);
+        final Bytes bytes = payload.payload().as();
+        out.writeInt((int) bytes.length());
+        bytes.writeTo(out);
     }
 
-    private void deserialize(
-            @NonNull final SerializableDataInputStream in, final int version, final int maxTransactionCount)
+    private static void serializeStateSignaturePayload(@NonNull final SerializableDataOutputStream out, @NonNull final EventPayload payload)
             throws IOException {
+        final StateSignaturePayload stateSignaturePayload = payload.payload().as();
+
+        out.writeLong(STATE_SIGNATURE_CLASS_ID);
+        out.writeInt(STATE_SIGNATURE_VERSION);
+        out.writeInt((int) stateSignaturePayload.signature().length());
+        stateSignaturePayload.signature().writeTo(out);
+
+        out.writeInt((int) stateSignaturePayload.hash().length());
+        stateSignaturePayload.hash().writeTo(out);
+
+        out.writeLong(stateSignaturePayload.round());
+        out.writeInt(Integer.MIN_VALUE); // epochHash is always null
+    }
+
+    /**
+     * Deserialize the event.
+     *
+     * @param in the stream from which this object is to be read
+     * @return the deserialized event
+     * @throws IOException if unsupported payload types are encountered
+     */
+    public static UnsignedEvent deserialize(@NonNull final SerializableDataInputStream in) throws IOException {
+        final int version = in.readInt();
+        if (version != ClassVersion.BIRTH_ROUND) {
+            throw new IOException("Unsupported version: " + version);
+        }
+
         Objects.requireNonNull(in, "The input stream must not be null");
-        softwareVersion = in.readSerializable(StaticSoftwareVersion.getSoftwareVersionClassIdSet());
+        final SoftwareVersion softwareVersion =
+                in.readSerializable(StaticSoftwareVersion.getSoftwareVersionClassIdSet());
 
-        creatorId = in.readSerializable(false, NodeId::new);
+        final NodeId creatorId = in.readSerializable(false, NodeId::new);
         if (creatorId == null) {
             throw new IOException("creatorId is null");
         }
-        selfParent = in.readSerializable(false, EventDescriptor::new);
-        otherParents = in.readSerializableList(AddressBook.MAX_ADDRESSES, false, EventDescriptor::new);
-        allParents = createAllParentsList();
-        birthRound = in.readLong();
+        final var selfParent = in.readSerializable(false, EventDescriptor::new);
```

Comment:
I only commented on the first one before, but there are still several instances of `var` in this file